### ISSUE TITLE
[Backup/Downloader] Option to auto-reinstall cogs from backup

### ISSUE THREAD

**Issue by Jackenmen** [Cog-Creators/Red-DiscordBot#2947]
# Feature request

#### Select the type of feature you are requesting:

- [ ] Cog
- [x] Command
- [ ] API functionality

#### Describe your requested feature

Currently `[p]backup` doesn't backup cogs from CogManager and RepoManager folder and backup repo urls in a `RepoManager/repos.json` file. It would be nice if the user would be able to automatically restore repos from this file and automatically install cogs that were previously installed.

Basically this is something that I want to see added to Core under `[p]restore` command or to Downloader under (I don't know what name of the command). If someone doesn't grab this, I eventually plan on writing this functionality but it may not happen any time soon. 
 
Related issue: #3421
